File: packages/rules/test/fixtures/twilio/py/missing-signature-verification.positive.py
from flask import Flask, request

app = Flask(__name__)


@app.route("/webhooks/twilio", methods=["POST"])
def twilio_webhook():
    # BUG: no RequestValidator, no manual HMAC.
    message_sid = request.form.get("MessageSid")
    print("Twilio event:", message_sid)
    return "", 200

File: packages/rules/test/fixtures/twilio/js/missing-signature-verification.positive.ts
import express from "express";

const app = express();
app.use(express.urlencoded({ extended: false }));

app.post("/webhooks/twilio", async (req, res) => {
  // BUG: no validateRequest, no manual HMAC; X-Twilio-Signature ignored.
  const event = req.body;
  console.log("Twilio event:", event.MessageSid);
  res.status(200).send("<Response/>");
});

app.listen(3000);
